src/services/mongo/pagination.ts

```diff
@@ -0,0 +1,42 @@
+export default class Pagination {
+    private readonly sortBy: string;
+    private readonly order: string;
+    private readonly page: number; // Starts from 1
+    private readonly pageSize: number;
+
+    // Defaults
+    private constructor(
+        sortBy = '_id',
+        order = 'ASC',
+        page = 1,
+        pageSize = 100,
+    ) {
+        this.sortBy = sortBy;
+        this.order = order;
+        this.page = page;
+        this.pageSize = pageSize;
+    }
+
+    private toFindOptions(): object {
+        return {
+            take: this.pageSize,
+            skip: ((this.page - 1) * this.pageSize),
+            order: {
+                [this.sortBy]: this.order,
+            },
+        };
+    }
+
+    public static create(
+        sortBy: string | undefined,
+        order: string | undefined,
+        page: number | undefined,
+        pageSize: number | undefined,
+    ): object {
+        if (sortBy === undefined && order === undefined && page === undefined && pageSize === undefined) {
```

Comment:
Hmmm not quite sure about this logic, wouldn't the Pagination class handle this?

Reply:
The constructor will default the rest if some of the parameters are undefined. But if none of them are specified I figured, we don't need to paginate at all (returns empty object).

Reply:
May as well have the object handle this so we don't need to wrap logic around it, plus it means we can typehint the pagination object in every case, even if it's empty.

Reply:
Good point. just took me awhile to figure out how to do this in typescript. I simplified the pagination, there are no default values now.
The only 3 properties are take, skip and order. Which are mapped from the query parameters: sortBy, order, page, pageSize. Those that are undefined will not get passed as FindOption when querying mongo.